draft-ietf-quic-transport.md

```diff
@@ -937,20 +937,17 @@ connection ID for the duration of the connection or until its peer invalidates
 the connection ID via a RETIRE_CONNECTION_ID frame
 ({{frame-retire-connection-id}}).
 
-Endpoints store received connection IDs for future use.  An endpoint that
-receives excessive connection IDs MAY discard those it cannot store.  An
-endpoint that issues connection IDs cannot expect its peer to store and use all
-issued connection IDs.
-
 An endpoint SHOULD ensure that its peer has a sufficient number of available and
-unused connection IDs.  While each endpoint independently chooses how many
-connection IDs to issue, endpoints SHOULD provide and maintain at least eight
-connection IDs.  The endpoint can do this by always supplying a new connection
-ID when a connection ID is retired by its peer or when the endpoint receives a
-packet with a previously unused connection ID.  Endpoints that initiate
-migration and require non-zero-length connection IDs SHOULD provide their peers
-with new connection IDs before migration, or risk the peer closing the
-connection.
+unused connection IDs. Endpoints store received connection IDs for future use.
+They advertise how many unretired connection IDs they are willing to store in
+the transport parameters. An endpoint MUST NOT provide more connection IDs than
+this limit. An endpoint MUST treat receipt of more than this number of
+connection IDs as an error of type CONNECTION_ID_LIMIT_ERROR. If an endpoint has
```

Comment:
Why is this error necessary? A peer can simply drop the ones that it can't store.

Reply:
I regard this as an equivalent to flow control and stream ID flow control. We have errors there if the peer violates a limit. It feels weird to me to set a limit, but then to silently tolerate a violation of that limit. And it will be a lot easier to develop compliant implementation if we actually enforce the limit.